### PR TITLE
Make Joker compile on Linux Debian with rtmidi from debian repo

### PR DESCRIPTION
**Pull request by TheErk**
Hi Martin,

I did give another try to make Linux compilation work on my Debian box.
here comes the patch set.
I did:
- Update README in order to give informations for building on Linux
- Tweak the  libs/PhMidi/* files so that it can compile on Debian Stretch with RtMidi coming from the repo.

This seems to work on my side.
Then I began to play with Joker for a sample video but I discovered I may have to write the detx flle manually? Am I missing something?

### CODE REVIEW

**Comment by MartinDelille** [comment]
Hi @TheErk!

Glad to see some Debian developer helping the project!
Your PR seems fine and I might update the RtMidi library in the future so we could remove the specific non linux code.

The appveyor build is failing but it is not related to you. I just push a fix that should make it green.

Can you rebase/merge the master branch and push? It'll trigger the travis-ci build that I deactivated accidentally for pull request. I just want to make sure it doesn't break anything.

**Comment by TheErk** [comment]
2016-06-22 17:26 GMT+02:00 Martin Delille <notifications@github.com>:

> Hi @TheErk <https://github.com/TheErk>!
>
> Glad to see some Debian developer helping the project!
> Your PR seems fine and I might update the RtMidi library in the future so
> we could remove the specific non linux code.
>
> The appveyor build is failing but it is not related to you. I just push a
> fix that should make it green.
>
> Can you rebase/merge the master branch and push? It'll trigger the
> travis-ci build that I deactivated accidentally for pull request. I just
> want to make sure it doesn't break anything.
>
I did update my master, rebase my branch on it and repush.

Is it ok for you now?


-- 
Eric

**Comment by TheErk** [comment]
Hi Martin,

I saw the Travis CI build failed but I'm not sure I understand why.
Shall I run uncrustify on the modified code ?

It is not clear from https://travis-ci.org/Phonations/Joker/builds/139604938
what are blocking errors?

**Comment by MartinDelille** [comment]
All failing build seems to be related to the pull request environment. I will investigate that. Regarding `uncrustify`, I didn't manage to have the same configuration for Mac and Linux so don't worry about it. If you know someone mastering it I'm interested but this is not a big issue anyway! 😃

File: appveyor.yml [hunk 1]
```diff
@@ -14,13 +14,13 @@ install:
...
+  - curl -kLO https://ffmpeg.zeranoe.com/builds/win32/dev/ffmpeg-20160622-e0faad8-win32-dev.zip
+  - 7z x ffmpeg-20160622-e0faad8-win32-dev.zip
+  - set FFMPEG_DEV_PATH=%CD%\ffmpeg-20160622-e0faad8-win32-dev
   # Installing ffmpeg shared
-  - curl -kLO http://ffmpeg.zeranoe.com/builds/win32/shared/ffmpeg-20151221-git-c0f67e1-win32-shared.7z
-  - 7z x ffmpeg-20151221-git-c0f67e1-win32-shared.7z
-  - set FFMPEG_SHARED_PATH=%CD%\ffmpeg-20151221-git-c0f67e1-win32-shared
+  - curl -kLO https://ffmpeg.zeranoe.com/builds/win32/shared/ffmpeg-20160622-e0faad8-win32-shared.zip
+  - 7z x ffmpeg-20160622-e0faad8-win32-shared.zip
+  - set FFMPEG_SHARED_PATH=%CD%\ffmpeg-20160622-e0faad8-win32-shared
```
Comment (MartinDelille) [inline appveyor.yml]: Please rebase to avoid these strange diffs.

Comment (TheErk) [inline appveyor.yml]: This diff is coming from your commit c3a91b346d367f85702d870ad24bf4f021eea5ae
commit c3a91b346d367f85702d870ad24bf4f021eea5ae
Author: Martin Delille <martin@phonations.com>
Date:   Wed Jun 22 17:23:11 2016 +0200

    update appveyor ffmpeg build

When I rebased my branch on Phonation/Joker tree I did inherit this commit.
Have a look here:
https://github.com/TheErk/Joker/commits/buildOnDebianLinux

File: libs/PhMidi/PhMidiInput.cpp [hunk 2]
```diff
@@ -61,7 +61,11 @@ bool PhMidiInput::open(QString inputPortName)
 			_midiIn->openVirtualPort(inputPortName.toStdString());
 		_midiIn->ignoreTypes( false, false, false );
 		_midiIn->setCallback(&PhMidiInput::callback, this);
+#ifndef  RTMIDI_PRE211_API
 		_midiIn->setErrorCallback(&PhMidiInput::errorCallback, this);
+#else
+                _midiIn->setErrorCallback(&PhMidiInput::errorCallback);
```
Comment (MartinDelille) [inline libs/PhMidi/PhMidiInput.cpp]: Use tabs here

File: libs/PhMidi/PhMidiInput.h [hunk 3]
```diff
@@ -150,7 +150,11 @@ private slots:
 
 private:
 	static void callback(double, std::vector< unsigned char > *message, void *userData );
+#ifndef RTMIDI_PRE211_API
 	static void errorCallback(RtMidiError::Type type, const std::string &errorText, void *userData);
+#else
+        static void errorCallback(RtMidiError::Type type, const std::string &errorText);
```
Comment (MartinDelille) [inline libs/PhMidi/PhMidiInput.h]: Use tab here

**Comment by MartinDelille** [comment]
It shouldn't appear then.

Could you please rebase your branch using the following commands?

    git rebase origin/master

**Pull request closed without merging**